StripePaymentSheet/StripePaymentSheet/Source/Internal/API Bindings/Link/STPAPIClient+Link.swift

```diff
@@ -448,8 +452,34 @@ extension STPPaymentMethodAddress {
 
     var consumersAPIParams: [String: Any] {
         var params = STPFormEncoder.dictionary(forObject: self)
+        // The param naming for consumers API is different so we need to map the values.
+        params["line_1"] = params["line1"]
+        params["line1"] = nil
+        params["line_2"] = params["line2"]
+        params["line2"] = nil
+        params["locality"] = params["city"]
+        params["city"] = nil
+        params["administrative_area"] = params["state"]
+        params["state"] = nil
         params["country_code"] = params["country"]
         params["country"] = nil
+
+        // Consumers API doesn't like empty strings, turn any empties into nils.
+        if let line1 = params["line_1"] as? String, line1.isEmpty {
+            params["line_1"] = nil
+        }
+        if let line2 = params["line_2"] as? String, line2.isEmpty {
+            params["line_2"] = nil
+        }
+        if let locality = params["locality"] as? String, locality.isEmpty {
+            params["locality"] = nil
+        }
+        if let administrativeArea = params["administrative_area"] as? String, administrativeArea.isEmpty {
+            params["administrative_area"] = nil
+        }
+        if let countryCode = params["country_code"] as? String, countryCode.isEmpty {
+            params["country_code"] = nil
+        }
```

Comment:
This would be a lot cleaner as a `map`, e.g.

```swift
let consumerKeyMap = [
  "line1": "line_1",
  "line2": "line_2",
  "city": "locality",
  "state": "administrative_area",
  "country": "country_code"
]

return STPFormEncoder.dictionary(forObject: self).map { key, value in
  let newKey = consumerKeyMap[key] ?? key
  let newValue: Any?

  // Consumers API doesn't like empty strings. Turn any empties into nils
  if let stringValue = value as? String, stringValue.isEmpty {
    newValue = nil
  } else {
    newValue = value
  }

  return (newKey, newValue)
}
```

Reply:
Nice one, didn't think of it